include/rmm/mr/device/thread_safe_resource_adaptor.hpp

```diff
@@ -35,7 +35,7 @@ namespace rmm::mr {
 template <typename Upstream>
 class thread_safe_resource_adaptor final : public device_memory_resource {
  public:
-  using lock_t = std::lock_guard<std::mutex>;
+  using lock_t = std::lock_guard<std::mutex>;  ///< Type of lock used to synchronize access
```

Comment:
Why were previous docs for this same using type just `RAII lock guard` ?

Reply:
Where do you see that? While an instance is used as a RAII lock guard, the type alias itself is not RAII.

Reply:
I think @wence- is referring to the other places in this PR where documentation was added that simply says `RAII lock guard`. But @harrism is correct about the distinction, I prefer this more explicit description and will apply it to the other places.

Reply:
I've made the change, but leaving this thread unresolved in case I've misunderstood the question.